<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GraphicsHardwareFixesSettingsTab</class>
 <widget class="QWidget" name="GraphicsHardwareFixesSettingsTab">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string/>
  </property>
  <layout class="QGridLayout" name="gridLayout" columnstretch="0,1">
   <item row="3" column="1">
    <widget class="QComboBox" name="hwAutoFlush">
     <item>
      <property name="text">
       <string>Disabled (Default)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Enabled (Sprites Only)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Enabled (All Primitives)</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="texInRTLabel">
     <property name="text">
      <string>Texture Inside RT:</string>
     </property>
     <property name="buddy">
      <cstring>textureInsideRt</cstring>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="skipDrawLabel">
     <property name="text">
      <string>Skip Draw Range:</string>
     </property>
     <property name="buddy">
      <cstring>skipDrawStart</cstring>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QComboBox" name="cpuCLUTRender">
     <property name="currentText">
      <string extracomment="0 (Disabled)">0 (Disabled)</string>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <item>
      <property name="text">
       <string>0 (Disabled)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>1 (Normal)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2 (Aggressive)</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="0" column="1">
    <layout class="QHBoxLayout" name="cpuCLUTRenderLayout" stretch="0,0">
     <item>
      <widget class="QComboBox" name="cpuSpriteRenderBW">
       <item>
        <property name="text">
         <string>0 (Disabled)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>1 (64 Max Width)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>2 (128 Max Width)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>3 (192 Max Width)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>4 (256 Max Width)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>5 (320 Max Width)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>6 (384 Max Width)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>7 (448 Max Width)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>8 (512 Max Width)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>9 (576 Max Width)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>10 (640 Max Width)</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="cpuSpriteRenderLevel">
       <item>
        <property name="text">
         <string>Sprites Only</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Sprites/Triangles</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Blended Sprites/Triangles</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item row="4" column="1">
    <widget class="QComboBox" name="textureInsideRt">
     <item>
      <property name="text">
       <string>Disabled (Default)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Inside Target</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Merge Targets</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="swCLUTLabel">
     <property name="text">
      <string>Software CLUT Render:</string>
     </property>
     <property name="buddy">
      <cstring>cpuCLUTRender</cstring>
     </property>
    </widget>
   </item>
   <item row="6" column="0" colspan="2">
    <layout class="QGridLayout" name="hwFixesLayout">
     <item row="4" column="0">
      <widget class="QCheckBox" name="estimateTextureRegion">
       <property name="text">
        <string>Estimate Texture Region</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QCheckBox" name="disableRenderFixes">
       <property name="text">
        <string>Disable Render Fixes</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QCheckBox" name="readTCOnClose">
       <property name="text">
        <string>Read Targets When Closing</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QCheckBox" name="disableSafeFeatures">
       <property name="text">
        <string>Disable Safe Features</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QCheckBox" name="disablePartialInvalidation">
       <property name="text">
        <string>Disable Partial Source Invalidation</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QCheckBox" name="disableDepthEmulation">
       <property name="text">
        <string>Disable Depth Conversion</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QCheckBox" name="frameBufferConversion">
       <property name="text">
        <string>Framebuffer Conversion</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QCheckBox" name="gpuPaletteConversion">
       <property name="text">
        <string>GPU Palette Conversion</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QCheckBox" name="preloadFrameData">
       <property name="text">
        <string>Preload Frame Data</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="hwAutoFlushLabel">
     <property name="text">
      <string>Auto Flush:</string>
     </property>
     <property name="buddy">
      <cstring>hwAutoFlush</cstring>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="gpuTargetCLUTLabel">
     <property name="text">
      <string extracomment="CLUT: Color Look Up Table, often referred to as a palette in non-PS2 things.  GPU Target CLUT: GPU handling of when a game uses data from a render target as a CLUT.">GPU Target CLUT:</string>
     </property>
     <property name="buddy">
      <cstring>gpuTargetCLUTMode</cstring>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QComboBox" name="gpuTargetCLUTMode">
     <item>
      <property name="text">
       <string>Disabled (Default)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Enabled (Exact Match)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Enabled (Check Inside Target)</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="cpuCLUTRenderLabel">
     <property name="text">
      <string>CPU Sprite Render Size:</string>
     </property>
     <property name="buddy">
      <cstring>cpuSpriteRenderBW</cstring>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <layout class="QHBoxLayout" name="skipDrawLayout">
     <item>
      <widget class="QSpinBox" name="skipDrawStart">
       <property name="maximum">
        <number>10000</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="skipDrawEnd">
       <property name="maximum">
        <number>10000</number>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="7" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>cpuSpriteRenderBW</tabstop>
  <tabstop>cpuSpriteRenderLevel</tabstop>
  <tabstop>cpuCLUTRender</tabstop>
  <tabstop>gpuTargetCLUTMode</tabstop>
  <tabstop>hwAutoFlush</tabstop>
  <tabstop>textureInsideRt</tabstop>
  <tabstop>skipDrawStart</tabstop>
  <tabstop>skipDrawEnd</tabstop>
  <tabstop>disableDepthEmulation</tabstop>
  <tabstop>frameBufferConversion</tabstop>
  <tabstop>disablePartialInvalidation</tabstop>
  <tabstop>gpuPaletteConversion</tabstop>
  <tabstop>disableSafeFeatures</tabstop>
  <tabstop>preloadFrameData</tabstop>
  <tabstop>disableRenderFixes</tabstop>
  <tabstop>readTCOnClose</tabstop>
  <tabstop>estimateTextureRegion</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
